waiter/src/waiter/scheduler/shell.clj

```diff
@@ -607,6 +658,27 @@
     (let [id->service @id->service-agent]
       (map (fn [[_ {:keys [service]}]] service) id->service)))
 
+  (signal-instance [this service-id instance-id signal-type timeout-ms] 
+   (if (scheduler/service-exists? this service-id)
+      (let [message (str "Sent " (name signal-type) " using scheduler API")
+            instance (get-in @id->service-agent [service-id :id->instance instance-id])]
+        (if (nil? instance)
+          (throw (IllegalArgumentException. "invalid instance-id"))) 
+        (if (get-in instance [:killed?])
+          (throw (IllegalArgumentException. "instance is already killed.")))
+        (if (and (not= signal-type :sigterm) (not= signal-type :sigkill))
+          (throw (IllegalArgumentException. "invalid signal type")))
+        (send id->service-agent send-signal-to-instance service-id instance-id message
+               port->reservation-atom port-grace-period-ms signal-type)
+        (scheduler/log-service-instance instance signal-type :info)
+        {:success true
+          :message (str signal-type " successfully sent to " instance-id)
+          :status http-200-ok})
+      {:success false
+       :message "service does not exist"
+       :status http-500-internal-server-error}))
```

Comment:
This is a client error as the service does not exist.
We should have a similar check for the instance and return a 404 in that scenario too.
```suggestion
       :status http-404-not-found}))
```